cache-config/t3c-apply/torequest/torequest.go

```diff
@@ -1030,6 +1034,45 @@ func (r *TrafficOpsReq) ProcessPackages() error {
 	return nil
 }
 
+func pkgMetaDataToMap(pmd []string) map[string]bool {
+	pkgMap := map[string]bool{}
+	for _, pkg := range pmd {
+		pkgMap[pkg] = true
+	}
+	return pkgMap
+}
+
+func pkgMatch(pkgMetaData []string, pk string) bool {
+	for _, pkg := range pkgMetaData {
+		if ok := strings.Contains(pk, pkg); ok {
+			return true
+		}
+	}
+	return false
+
+}
+
+func (r *TrafficOpsReq) ProcessPackagesWithMetaData(packageMetaData []string) error {
+	pkgs, err := getPackages(r.Cfg)
+	pkgMdataMap := pkgMetaDataToMap(packageMetaData)
+	if err != nil {
+		return errors.New("getting packages: " + err.Error())
```

Comment:
constructing errors from errors should wrap them e.g. using `fmt.Errorf`'s `%w` formatting directive so that the identity of the underlying error is not destroyed.

Reply:
updated `return fmt.Errorf("getting packages: %w", err)`